test/integration/oidc/oidc_test.go

```diff
@@ -0,0 +1,435 @@
+package oidc
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	rbacV1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
+	restClient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd/api"
+	kubeAPIServerTesting "k8s.io/kubernetes/cmd/kube-apiserver/app/testing"
+	"k8s.io/kubernetes/pkg/apis/rbac"
+	"k8s.io/kubernetes/pkg/kubeapiserver/authorizer/modes"
+	"k8s.io/kubernetes/test/integration/framework"
+	oidcTest "k8s.io/kubernetes/test/utils/oidc"
```

Comment:
Do not use caps in imports.  Use the last `N` segments to get unique names if needed such as `rbacv1`, `kubeapiserver/app/testing`, `utilsoidc`, etc